### PR TITLE
enh(unattended): update version and password generation for 22.04

### PR DESCRIPTION
**Pull request by tuntoja**
## Description

Updated version to 22.04
Updated password generation to be compliant with the new security policy.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: unattended.sh [hunk 1]
```diff
@@ -26,18 +26,22 @@ wizard_autoplay=${ENV_WIZARD_AUTOPLAY:-"false"} #Default the install wizard is n
 central_ip=${ENV_CENTRAL_IP:-$default_ip}       #Default central ip is the first of hostname -I
 
 function genpasswd() {
-	local _pwd
-	_pwd=$(date +%s%N | sha256sum | base64 | head -c 32)
+        local _pwd
```
Comment (Nohzoh) [inline unattended.sh:29]: we should always keep the same indentation
```suggestion
  local _pwd
```

File: unattended.sh [hunk 2]
```diff
@@ -26,18 +26,22 @@ wizard_autoplay=${ENV_WIZARD_AUTOPLAY:-"false"} #Default the install wizard is n
...
 
-	echo "Random password generated for user [$1] is [$_pwd]" >>$tmp_passwords_file
+        PWD_STRING_LENGTH=16
+        PWD_SPECIAL_CHARS='a-zA-Z0-9\!\@\$\*\?'
```
Comment (Nohzoh) [inline unattended.sh:32]: we should always keep the same indentation
```suggestion
  PWD_STRING_LENGTH=16
  PWD_SPECIAL_CHARS='a-zA-Z0-9\!\@\$\*\?'
```

File: unattended.sh [hunk 3]
```diff
@@ -26,18 +26,22 @@ wizard_autoplay=${ENV_WIZARD_AUTOPLAY:-"false"} #Default the install wizard is n
...
-		exit 1
-	fi
+        _pwd=$(cat /dev/urandom | tr -dc $PWD_SPECIAL_CHARS | fold -w $PWD_STRING_LENGTH | head -n 1)
+
+        echo "Random password generated for user [$1] is [$_pwd]" >>$tmp_passwords_file
+
+        if [ $? -ne 0 ]; then
+                echo "ERROR : Cannot save the random password to [$tmp_passwords_file]"
+                exit 1
+        fi
```
Comment (Nohzoh) [inline unattended.sh:41]: we should always keep the same indentation
```suggestion
  _pwd=$(cat /dev/urandom | tr -dc $PWD_SPECIAL_CHARS | fold -w $PWD_STRING_LENGTH | head -n 1)

  echo "Random password generated for user [$1] is [$_pwd]" >>$tmp_passwords_file

  if [ $? -ne 0 ]; then
    echo "ERROR : Cannot save the random password to [$tmp_passwords_file]"
    exit 1
  fi
```

File: unattended.sh [hunk 4]
```diff
@@ -26,18 +26,22 @@ wizard_autoplay=${ENV_WIZARD_AUTOPLAY:-"false"} #Default the install wizard is n
...
 
-	#return the generated password
-	echo $_pwd
+        #return the generated password
+        echo $_pwd
```
Comment (Nohzoh) [inline unattended.sh:44]: we should always keep the same indentation
```suggestion
  #return the generated password
  echo $_pwd
```

**Review by Nohzoh — CHANGES_REQUESTED** [review]

**Review by Nohzoh — APPROVED** [review]

**Review by quanghungb — APPROVED** [review]

**Pull request merged**